assets/model_monitoring/components/src/genai_token_statistics_compute_metrics/metrics_processor.py

```diff
@@ -54,127 +38,103 @@ def create_span_tree_from_dataframe_row(self, row: Row) -> Optional[SpanTreeNode
     def create_span_tree_from_dataframe(self, df: DataFrame) -> List[Optional[SpanTreeNode]]:
         """Create a list of SpanTrees from a DataFrame."""
         # Convert the DataFrame to an RDD and then to a list of Rows
-        rows = df.rdd.map(lambda row: row).collect()
-
-        # Create a SpanTree for each row
-        span_trees = [self.create_span_tree_from_dataframe_row(row) for row in rows]
-        return span_trees
+        df.rdd.foreach(self.process_row)
```

Comment:
I don't see anything returned, according to the signature, there should be something returned